lib/plugins/place_block.js

```diff
@@ -13,8 +13,12 @@ function inject (bot) {
       [oldBlock, newBlock] = await onceWithCleanup(bot, `blockUpdate:${dest}`, { timeout: 5000 })
     }
 
-    if (oldBlock.type === newBlock.type) {
-      throw new Error(`No block has been placed : the block is still ${oldBlock.name}`)
+    // blockUpdate emits (null, null) when the world unloads
+    if (!oldBlock && !newBlock) {
+      return
+    }
+    if (oldBlock?.type === newBlock?.type) {
```

Comment:
Can new block actually be undefined?

Reply:
Yes. When you register a listener for block updates and the world unloads you receive a block update even with both new block and old block being null. https://github.com/PrismarineJS/mineflayer/blob/8f87e055d9dc6ab58e2efff036be589b3086b970/lib/plugins/blocks.js#L481

Reply:
Actually this looks like it only applies to block update events tied to specific coordinates. The generic blockUpdate event can indeed not have null for newBlock. Ill fix that.